public/scripts/project/project.js

```diff
@@ -103,6 +104,17 @@ define(function(require) {
 
     var metadataLocation = _user && _anonymousId ? Path.join(Constants.ANONYMOUS_USER_FOLDER, _anonymousId.toString()) : getRoot();
 
+    // First, lock the project or bail
+    var locked = Lock.setLock(_id);
+    if (!locked) {
+      // Abort!
+      window.location = "/projects";
+    }
+
+    // Remove the lock when the user navigates away from the page, or closes
+    // the tab.
+    window.onbeforeunload = Lock.removeLock.bind(Lock, _id);
```

Comment:
you don't need to bind anything here, just call it as is. Also, use `window.addEventListener("beforeunload")`